apps/dashboard/src/components/primitives/button.tsx

```diff
@@ -43,12 +44,34 @@ export interface ButtonProps
   extends React.ButtonHTMLAttributes<HTMLButtonElement>,
     VariantProps<typeof buttonVariants> {
   asChild?: boolean;
+  isLoading?: boolean;
 }
 
 const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
-  ({ className, variant, size, asChild = false, ...props }, ref) => {
+  ({ className, variant, size, asChild = false, isLoading = false, children, disabled, ...props }, ref) => {
     const Comp = asChild ? Slot : 'button';
-    return <Comp className={cn(buttonVariants({ variant, size, className }))} ref={ref} {...props} />;
+    return (
+      <Comp
+        className={cn(buttonVariants({ variant, size, className }), isLoading && 'animate-pulse-subtle')}
+        ref={ref}
+        disabled={disabled || isLoading}
+        {...props}
+      >
+        <span
+          className={cn(
+            'flex items-center gap-1 transition-all duration-300',
```

Comment:
We want to keep the text, but hide using opacity to preserve button width

Reply:
Nice!